Tests/CalculatorApp.Tests/OrderOfOperationsTests.cs

```diff
@@ -0,0 +1,119 @@
+using Xunit;
+
+namespace CalculatorApp.Tests;
+
+public class OrderOfOperationsTests
+{
+    private readonly Calculator _calculator;
+
+    public OrderOfOperationsTests()
+    {
+        _calculator = new Calculator();
+    }
+
+    [Theory]
+    [InlineData("2+3*4", 14)]  // 2 + (3*4) = 2 + 12 = 14
+    [InlineData("10-2*3", 4)] // 10 - (2*3) = 10 - 6 = 4
+    [InlineData("20/4+2", 7)] // (20/4) + 2 = 5 + 2 = 7
+    [InlineData("3*4+5", 17)] // (3*4) + 5 = 12 + 5 = 17
+    // [InlineData("-2+3", 1)]  // negation has higher precedence than addition: (-2) + 3 = 1
+    // [InlineData("-2*3", -6)] // negation has higher precedence than multiplication: (-2) * 3 = -6
+    public void MultiplicationDivision_OverAdditionSubtraction_ShouldReturnCorrectResult(string expression, double expected)
+    {
+        var result = _calculator.Evaluate(expression);
+        Assert.Equal(expected, result, 10);
+    }
+
+    [Theory]
+    [InlineData("2^3*4", 32)] // (2^3) * 4 = 8 * 4 = 32
+    [InlineData("3*2^2", 12)] // 3 * (2^2) = 3 * 4 = 12
+    [InlineData("2^2+3^2", 13)] // (2^2) + (3^2) = 4 + 9 = 13
+    [InlineData("10-2^3", 2)] // 10 - (2^3) = 10 - 8 = 2
+    // [InlineData("-2^2", -4)] // negation has lower precedence than exponentiation: -(2^2) = -4
+    // [InlineData("(-2)^2", 4)] // parentheses override precedence: (-2)^2 = 4
+    public void Exponentiation_OverMultiplicationDivision_ShouldReturnCorrectResult(string expression, double expected)
+    {
+        var result = _calculator.Evaluate(expression);
+        Assert.Equal(expected, result, 10);
+    }
+
+    [Theory]
+    [InlineData("2+3*4-5", 9)]   // 2 + (3*4) - 5 = 2 + 12 - 5 = 9
+    [InlineData("10/2+3*4", 17)] // (10/2) + (3*4) = 5 + 12 = 17
+    [InlineData("2^2*3+1", 13)]  // (2^2) * 3 + 1 = 4 * 3 + 1 = 13
+    [InlineData("20-3*2^2", 8)]  // 20 - (3 * (2^2)) = 20 - (3 * 4) = 20 - 12 = 8
+    // [InlineData("-2^2+3", -1)] // negation, then exponentiation, then addition: -(2^2) + 3 = -4 + 3 = -1
```

Comment:
actually exponentiation comes before negation

```suggestion
    // [InlineData("-2^2+3", -1)] // exponentiation, then negation, then addition: -(2^2) + 3 = -4 + 3 = -1
```